### PR TITLE
Add UseGalaxy Help Forum to Matrix channel Bot

### PR DESCRIPTION
**Pull request by mtekman**
This is a Matrix bot written in Bash to fetch posts from https://help.usegalaxy.org that match certain tags, filter for less than N replies and post a nicely formatted table to a Matrix room of your choice.

Inspired by a great idea from @pavanvidem and @nomadscientist , and inspired further by @hexylena's Ruby bot.

I'm not sure if I'm adding it to the right location in the file system, or how/where to deploy it (other than running it from my own account on my own system), but here goes.

### CODE REVIEW

**Review by hexylena — COMMENTED** [review]
in bash! hilarious :)

I feel like it's a serialisation issue waiting to happen, but it's also not the worst if it fails for some reason. (and we'll just ping you to fix it after all). But otherwise this is pretty clean bash, I don't have any major issues with it. You could consider adding `jq` into the pipeline to help build the json, but that's about as painful as rewriting it in python/perl/ruby/any language with a proper json library/hash implementation.

The deployment is via a github action, could you please copy .github/workflows/matrix.yml into something for your script?

**Comment by mtekman** [comment]
Yeah I was planning on shoving that final json through jq, but it kept turning my `\"` into `"` which annoyed me, so I left it out.

Still, I should at least test to see if it's a valid JSON and bail if not

**Comment by hexylena** [comment]
Ping me when ya add the action (based on `.github/workflows/matrix.yml` is fine) and I'll merge it :)

**Comment by mtekman** [comment]
@hexylena ping! Though is there any way to check if the `MATRIX_ACCESS_TOKEN=${MATRIX_ACCESS_TOKEN}` line is needed? I guess it's already in the env and won't need to be re-set?

Also, I guess I need to add the GTN bot account to the room

**Comment by hexylena** [comment]
> MATRIX_ACCESS_TOKEN=${MATRIX_ACCESS_TOKEN}

should be redundant/nop, but yea `MATRIX_ACCESS_TOKEN` is the value set in github actino's config.

> Also, I guess I need to add the GTN bot account to the room

yeah it's probably easier that way. I can log into the bot account and accept it, I haven't figured out how to manage that interaction yet (if you figure it out tho that'd be ace)

**Comment by mtekman** [comment]
Okay I'll remove the `MATRIX_ACCESS_TOKEN` in a later PR once I know the bot's working

>  I can log into the bot account and accept it, I haven't figured out how to manage that interaction yet (if you figure it out tho that'd be ace)

I just sent an invite if that helps

**Comment by hexylena** [comment]
Accepted

File: .github/workflows/scrna-galaxyhelp-matrix.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y coreutils curl libxml2-utils gawk sed grep jq wget
+      # END Dependencies
+
+      - name: Send announcements to single-cell matrix for unanswered posts
+        run: |
+            wget https://raw.githubusercontent.com/galaxyproject/training-material/main/bin/galaxy-help-news.sh
+            MATRIX_ACCESS_TOKEN="${MATRIX_ACCESS_TOKEN}" HTML_TYPE="bullets" MAX_REPLIES="0" WANTED_TAGS="scrna scrna-seq" ROOM_ID='!TJRLNvfcbWbSRoUNpl:matrix.org' bash ./galaxy-help-news.sh
```
Comment (hexylena) [inline .github/workflows/scrna-galaxyhelp-matrix.yml:30]:
```suggestion
            HTML_TYPE="bullets" MAX_REPLIES="0" WANTED_TAGS="scrna scrna-seq" ROOM_ID='!TJRLNvfcbWbSRoUNpl:matrix.org' bash ./galaxy-help-news.sh
```

yeah not necessary.

**Review by hexylena — DISMISSED** [review]
Very cool. If this works I think it might be interesting for the admins too, we don't see those posts until @jennaj mentions them to us in the matrix channel

**Review by hexylena — APPROVED** [review]

**Pull request merged**